tests/jballerina-debugger-integration-test/src/test/resources/project-based-tests/variable-tests-2/main.bal

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+type Person record {|
+    int id;
+    string fname;
+    string lname;
+|};
+
+type SampleErrorData record {|
+    int code;
+    string reason;
+|};
+
+type SampleError error<SampleErrorData>;
+
+public function main() {
```

Comment:
Shall we have a test for binding patterns in a foreach also (It is the originally created issue right?)?

Reply:
Yes, agreed. Added the requested tests in https://github.com/ballerina-platform/ballerina-lang/pull/43624/commits/25415e4672032616bbbf844c87510107e1983ffd